Source/WebCore/dom/Document.cpp

```diff
@@ -8933,6 +8933,66 @@ HTMLDialogElement* Document::activeModalDialog() const
     return nullptr;
 }
 
+HTMLElement* Document::topmostAutoPopover() const
+{
+    for (auto& element : makeReversedRange(m_topLayerElements)) {
+        if (auto* candidate = downcast<HTMLElement>(element.ptr()); candidate->popoverState() == PopoverState::Auto)
+            return candidate;
+    }
```

Comment:
If there is no guarantee these are HTML elements, then we need a dynamic downcast:
```suggestion
    for (auto& element : makeReversedRange(m_topLayerElements)) {
        if (auto* candidate = dynamicDowncast<HTMLElement>(element.get()); candidate->popoverState() == PopoverState::Auto)
            return candidate;
    }
```